### PR TITLE
Change message of `invalid_iboutlet` and `invalid_ibinspectable` to reflect accurate usage

### PR DESCRIPTION
**Pull request by NSAntoine**
The current error message for when `@GKInspectable`, `@IBInspectable`, and `@IBOutlet` is `only instance properties can be declared @GKInspectable / @IBInspectable / @IBOutlet`, however it doesn't reflect that it should be an instance property of a class, just that it should be an instance property, though those attributes are only allowed to be used for instance properties of a class type.

This pull request changes the message so that it should reflect the attribute should be used on a class instance property.

### CODE REVIEW

**Comment by SerenaKit** [comment]
> Looks like those two error macros are pretty much identical, and we can get rid of the one that hardcodes `IBOutlet` in favor of the other one that takes a string argument, and pass `attr->getName` for `IBOutletAttr`. Do you mind cleaning that up as well?

I noticed this yesterday and wanted to replace the hardcoded one but thought maybe I shouldn’t overcomplicate the pr but it should be fine now that I think about it, I’ll do it once I’m near my computer again

**Comment by SerenaKit** [comment]
@AnthonyLatsis

**Review by AnthonyLatsis — COMMENTED** [review]
Looks great, thank you!

File: lib/Sema/TypeCheckAttr.cpp [hunk 1]
```diff
@@ -632,15 +632,15 @@ void AttributeChecker::visitIBInspectableAttr(IBInspectableAttr *attr) {
   // Only instance properties can be 'IBInspectable'.
   auto *VD = cast<VarDecl>(D);
   if (!VD->getDeclContext()->getSelfClassDecl() || VD->isStatic())
-    diagnoseAndRemoveAttr(attr, diag::invalid_ibinspectable,
+    diagnoseAndRemoveAttr(attr, diag::attr_must_be_used_on_class_instance,
                                  attr->getAttrName());
```
[the remark this answers is not in the document]
Comment (SerenaKit) [reply lib/Sema/TypeCheckAttr.cpp:636]: Fixed!

**Comment by SerenaKit** [comment]
@AnthonyLatsis may be redundant but shouldn’t we run tests just one more time to make sure nothing got messed up?

**Comment by AnthonyLatsis** [comment]
Sure.

@swift-ci please smoke test macOS

**Review by hborla — APPROVED** [review]
Nice improvement!

**Comment by hborla** [comment]
@swift-ci please smoke test

**Comment by AnthonyLatsis** [comment]
@SerenaKit Before we merge, could you rebase this into 2 commits, one for the message edit and another one for the gardening? Would also be nice if the second commit message was a tad more specific.

**Comment by SerenaKit** [comment]
@AnthonyLatsis There, should be fine now (I think)

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test and merge

**Pull request merged**